coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -267,6 +269,19 @@ class QueryEngine(dataset: Dataset,
     }
   }
 
+  private def materializeApplyMiscellaneousFunction(queryId: String,
+                                              submitTime: Long,
+                                              options: QueryOptions,
+                                              lp: ApplyMiscellaneousFunction): Seq[ExecPlan] = {
+    val vectors = walkLogicalPlanTree(lp.vectors, queryId, submitTime, options)
+    lp.function match {
+      case LabelReplace =>
+        vectors.foreach(_.addRangeVectorTransformer(exec.LabelFunctionMapper(lp.function,lp.functionArgs)))
```

Comment:
Consider having one RangeVectorTransformer with all the misc functions. The transformer (aka mapper) could perform the validation and apply of all functions.

Reply:
Done